src/wallet/wallet.cpp

```diff
@@ -508,30 +508,39 @@ void CWallet::SetBestChain(const CBlockLocator& loc)
     SetBestChainINTERNAL(walletdb, loc);
 }
 
-std::set<std::pair<libzcash::PaymentAddress, uint256>> CWallet::GetNullifiersForAddresses(const std::set<libzcash::PaymentAddress> & addresses)
+std::set<std::pair<libzcash::PaymentAddress, uint256>> CWallet::GetNullifiersForAddresses(
+        const std::set<libzcash::PaymentAddress> & addresses)
 {
     std::set<std::pair<libzcash::PaymentAddress, uint256>> nullifierSet;
-    // Sapling ivk -> addr map
-    std::map<libzcash::SaplingIncomingViewingKey, libzcash::SaplingPaymentAddress> ivkMap;
+    // Sapling ivk -> list of addrs map
+    std::map<libzcash::SaplingIncomingViewingKey, std::vector<libzcash::SaplingPaymentAddress>> ivkMap;
     for (const auto & addr : addresses) {
-        if (boost::get<libzcash::SaplingPaymentAddress>(&addr) != nullptr) {
-            auto saplingAddr = boost::get<libzcash::SaplingPaymentAddress>(addr);
+        auto saplingAddr = boost::get<libzcash::SaplingPaymentAddress>(&addr);
+        if (saplingAddr != nullptr) {
             libzcash::SaplingIncomingViewingKey ivk;
-            this->GetSaplingIncomingViewingKey(saplingAddr, ivk);
-            ivkMap.insert(std::make_pair(ivk, saplingAddr));
+            this->GetSaplingIncomingViewingKey(*saplingAddr, ivk);
+            ivkMap[ivk].push_back(*saplingAddr);
```

Comment:
pretty cool how this constructs the empty vector on the first reference to `ivkMap[]`